File: linkedlist/21_MergeTwoSortedLists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
 var mergeTwoLists = function(list1, list2) {
    var head;
    var node;
    console.log(list1)
    while(list1 != null || list2 != null){

        let data1 = list1?.val ?? 101;
        let data2 = list2?.val ?? 101;

        let pick = -101;

        if(data1 > data2){
            pick = data2;
            list2 = list2.next
        }
        else{
            pick = data1;
            list1 = list1.next
        }

        if(head == null){
            node = new ListNode(pick)
            head = node
        }
        else{
            node.next = new ListNode(pick)
            node = node.next
        }
    }
    if(head == null)
        head = ListNode()
    return head;
};

File: linkedlist/21_MergeTwoSortedLists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {

	var head *ListNode
	var node *ListNode

	for list1 != nil || list2 != nil {

		data1 := 101
		data2 := 101
		pick := 101

		if list1 != nil {
			data1 = list1.Val
		}

		if list2 != nil {
			data2 = list2.Val
		}

		if data1 > data2 {
			pick = data2
			list2 = list2.Next
		} else {
			pick = data1
			list1 = list1.Next
		}

		if head == nil {
			node = new(ListNode)
			node.Val = pick
			head = node
			continue
		}

		node.Next = new(ListNode)
		node = node.Next
		node.Val = pick

	}

	if head == nil {
		return list1
	}

	return head
}